engine/table/src/test/java/io/deephaven/engine/table/impl/lang/TestQueryLanguageParser.java

```diff
@@ -1798,29 +2307,32 @@ public void testGenerics() throws Exception {
         resultExpression = "genericDoubleArrayToSingle(new Double[0][0])";
         check(expression, resultExpression, Double.class, new String[] {});
 
-        expression = "genericGetKey(myHashMap)";
-        resultExpression = "genericGetKey(myHashMap)";
-        check(expression, resultExpression, Integer.class, new String[] {"myHashMap"});
+        expression = "genericGetKey(myParameterizedHashMap)";
+        resultExpression = "genericGetKey(myParameterizedHashMap)";
+        check(expression, resultExpression, Integer.class, new String[] {"myParameterizedHashMap"});
 
-        expression = "genericGetValue(myHashMap)";
-        resultExpression = "genericGetValue(myHashMap)";
-        check(expression, resultExpression, Double.class, new String[] {"myHashMap"});
+        expression = "genericGetValue(myParameterizedHashMap)";
+        resultExpression = "genericGetValue(myParameterizedHashMap)";
+        check(expression, resultExpression, Double.class, new String[] {"myParameterizedHashMap"});
     }
 
     public void testVectorUnboxing() throws Exception {
         String expression = "genericArrayToSingle(myVector)";
         String resultExpression = "genericArrayToSingle(VectorConversions.nullSafeVectorToArray(myVector))";
-        check(expression, resultExpression, Double.class, new String[] {"myVector"});
+        // TODO: when reparsing result, VectorConversions.nullSafeVectorToArray(myVector) is seen as a java.lang.Double
```

Comment:
I just fixed it instead; it was an issue where `<T> T[] myMethod(T arg)` ignored the array-ness of the return type (so `myMethod(myDoubleObj)` was returning `T` instead of `T[]`).